network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt

```diff
@@ -0,0 +1,45 @@
+package com.halcyonmobile.android.common.extensions.network
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+/**
+ * Checks if the device's active Network has internet capability.
+ *
+ * @return true if Device has Active Internet Available Network, false otherwise.
+ */
+val Context.isNetworkActive: Boolean
+    get() {
+        val connectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        return connectivityManager.isNetworkActiveCompat
+    }
+
+/**
+ * Checks if the device's active Network has internet capability.
+ *
+ * @return true if Device has Active Internet Available Network, false otherwise.
+ */
+val ConnectivityManager.isNetworkActiveCompat: Boolean
+    get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        isNetworkActive
+    } else {
+        isNetworkActiveDeprecated
+    }
+
+@get:RequiresApi(Build.VERSION_CODES.M)
+val ConnectivityManager.isNetworkActive: Boolean
```

Comment:
It seems to me that we could hide this, what do you think?

Reply:
My thinking here was that, if your min API is above 23, you wouldn't use the compat + deprecated things anymore, just copy this instead